tests/test_bidsRun.py

```diff
@@ -0,0 +1,163 @@
+import logging
+
+import nibabel as nib
+import numpy as np
+import pytest
+
+from rtCommon.bidsCommon import getNiftiData
+from rtCommon.bidsIncremental import BidsIncremental
+from rtCommon.bidsRun import BidsRun
+from rtCommon.errors import MetadataMismatchError
+
+logger = logging.getLogger(__name__)
+
+
+# Test equality check is correct
+def testEq(oneImageBidsI):
+    run1 = BidsRun()
+    run2 = BidsRun()
+
+    run1.appendIncremental(oneImageBidsI)
+    assert run1 != run2
+
+    run2.appendIncremental(oneImageBidsI)
+    assert run1 == run2
+
+    run2._entities['subject'] = "new_subect"
+    assert run1 != run2
+
+
+# Test numIncrementals output is correct
+def testNumIncrementals(oneImageBidsI):
+    run = BidsRun()
+    assert run.numIncrementals() == 0
+
+    NUM_APPENDS = 20
+    for i in range(NUM_APPENDS):
+        run.appendIncremental(oneImageBidsI)
```

Comment:
It would be good if incrementals internally kept track their index number so trying to append the same incremental muiltiple times would be an error (in case for example a network message arrived twice). But that can be a TODO for later.

Reply:
Agreed, my thought was also to leave it as a TODO for later. I'll create an issue (feature enhancement) to formalize it.